x/pairing/keeper/scores/score.go

```diff
@@ -68,19 +69,23 @@ func init() {
 	}
 }
 
-// get the overall requirements from the policy and assign slots that'll fulfil them
-// TODO: this function should be changed in the future since it only supports stake reqs
+// CalcSlots gets the overall requirements from the policy and assign slots that'll fulfil them
 func CalcSlots(policy planstypes.Policy, minStake sdk.Int) []*PairingSlot {
-	// init slot array (should be as the number of providers to pair)
 	slots := make([]*PairingSlot, policy.MaxProvidersToPair)
 
-	// all slots should consider the stake, so we init them with stakeReq
+	// stake requirements
 	stakeReq := StakeReq{MinStake: minStake}
 	stakeReqName := stakeReq.GetName()
 
+	// geo requirements
+	geoReqsForSlots := GetGeoReqsForSlots(policy)
```

Comment:
this is dangerous, the list here HAVE to be in the length of slots.
can you make this function GetGeoReqsForSlot that also recieve the slot index and do it in the loop?

Reply:
done. see https://github.com/lavanet/lava/commit/76f0ff8449b77d96bffbe1f5a06fcb1eca13f9e8